File: app/src/main/java/me/cafecode/android/newspaper/injection/AppModule.java
package me.cafecode.android.newspaper.injection;

import dagger.Module;
import dagger.Provides;
import me.cafecode.android.newspaper.data.NewsRepository;

/**
 * Created by Natthawut Hemathulin on 5/31/2017 AD.
 * Email: natthawut1991@gmail.com
 */

@Module
public class AppModule {

    public AppModule() {

    }

    @Provides
    NewsRepository provideRepository() {
        return new NewsRepository();
    }

}

File: app/src/main/java/me/cafecode/android/newspaper/injection/AppModule.kt
package me.cafecode.android.newspaper.injection

import dagger.Module
import dagger.Provides
import me.cafecode.android.newspaper.util.BaseSchedulerProvider
import me.cafecode.android.newspaper.util.SchedulerProvider

@Module
class AppModule {

    @Provides
    fun provideScheduleProvider(): BaseSchedulerProvider {
        return SchedulerProvider()
    }

}